frontend/src/metabase/common/components/EntityPicker/components/EntityPickerModal/EntityPickerModal.tsx

```diff
@@ -83,6 +83,23 @@ export function EntityPickerModal<TItem extends TypeWithModel>({
   const hasTabs = tabs.length > 1 || searchQuery;
   const tabModels = useMemo(() => tabs.map(t => t.model), [tabs]);
 
+  useEffect(() => {
+    const handleEscapeModal = (event: KeyboardEvent) => {
+      if (event.key === "Escape") {
+        event.stopPropagation();
+        onClose();
+      }
+    };
+
+    document.addEventListener("keydown", handleEscapeModal, true);
```

Comment:
I do not love attaching this event listener to the document, or forcing it to listen on the capture phase, but I couldn't figure out any more constrained way to keep it from preventing propagation to other modals.

Reply:
If I understand right, Mantine's `onClose` prop handles the modal's closure logic. Is `onClose` called twice now, once by `handleEscapeModal` and once by Mantine?

Reply:
@iethree I'll approve and let you decide whether this is a concern